test/literal/enum.test.rb

```diff
@@ -10,15 +10,21 @@
 end
 
 Switch = Literal::Enum(_Boolean).define do
-	On(true)
-	Off(false)
+	On(true) do
+		def toggle = Switch::Off
```

Comment:
Unfortunately, these need to be qualified with `Switch::` because we're using block-based definitions.

The reason we're using block-based definitions is we need to run some code after the definition in order to freeze the enum class.

An alternative is to use a `TracePoint` to detect the end of the class definition, which would allow us to do this:

```ruby
class Switch < Literal::Enum[_Boolean]
  On(true) do
    def toggle = Off
  end

  Off(false) do
    def toggle = On
  end
end
```

Reply:
I personally prefer this interface, but I don't want to use `TracePoint` too much. Our other uses of `TracePoint` — such as for abstract methods — are non-essential and can be disabled.